s2i-liberty/readme.md

```diff
@@ -0,0 +1,197 @@
+	* [Environment Setup](#environment-setup)
+	* [Produce Runtime Image](#produce-runtime-image)
+		* [Git Source](#git-source)
+		* [Binary Source](#binary-source)
+	* [Create the build config](#create-the-build-config)	
+	* [Create A New Application](#create-a-new-application)
+    * [Considerations on HTTP session failover](#considerations-on-http-session-failover)	
+
+
+## Overview
+
+OpenShift provides several out of the box Source to Image builder images. To support deployments in Liberty, a new s2i builder image will be created to support a simplified deployment to OpenShift. Once the new image is produced, an example application will be deployed. 
+
+We will use the [extended builds approach](https://docs.openshift.org/latest/dev_guide/builds.html#extended-builds).
+
+With this approach we decouple the build phase and the builder image from the run phase and the runtime image. The builder image is still used to run the builds but the resulting artifact are passed to the runtime image so that it can create a final runnable image. This is all handled by OpneShift with few simple configurations.  
+
+## Bill of Materials
+
+### Environment Specifications
+
+This demo should be run on an installation of OpenShift Enterprise V3.3
+
+### Template Files
+
+None
+
+### Config Files
+
+None
+
+### External Source Code Repositories
+
+* [Example Jee Application](https://github.com/efsavage/hello-world-war) -  Public repository for a simple jee hello world app.
+
+## Setup Instructions
+
+There is no specific requirements necessary for this demonstration. The presenter should have an OpenShift Enterprise 3.3 environment available with access to the public Internet and the OpenShift Command Line Tools installed on their machine.
+
+## Presenter Notes
+
+The following steps are to be used to demonstrate two methods for producing the Source to Image builder image and deploying an application using the resulting image.
+
+### Environment Setup
+
+Using the OpenShift CLI, login to the OpenShift environment.
+
+
+```
+oc login <OpenShift_Master_API_Address>
+```
+
+Create a new project called *play-demo*
+
+
+```
+oc new-project liberty-demo
+```
+
+### Produce Runtime Image
+
+The Liberty runtime image can be created using the [Git](https://docs.openshift.com/enterprise/latest/dev_guide/builds.html#source-code) or [Binary](https://docs.openshift.com/enterprise/latest/dev_guide/builds.html#binary-source) build source
+
+In this example we will use a pre-existing builder image that can build maven based apps: `registry.access.redhat.com/jboss-eap-7/eap70-openshift` 
+
+#### Git Source
+
+The content used to produce the Liberty runtime image can originate from a Git repository. Execute the following command to start a new image build using the git source strategy.:
+
+```
+oc new-build websphere-liberty:webProfile7~https://github.com/raffaelespazzoli/containers-quickstarts --context-dir=s2i-liberty --name=liberty-runtime-s2i --strategy=docker
```

Comment:
let's change the git repo name to `https://github.com/redhat-cop/containers-quickstarts`